packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -3,18 +3,68 @@
 // found in the LICENSE file.
 
 import 'dart:math' show min, max;
-import 'dart:ui' as ui show Paragraph, ParagraphBuilder, ParagraphConstraints, ParagraphStyle;
+import 'dart:ui' as ui show Paragraph, ParagraphBuilder, ParagraphConstraints, ParagraphStyle, Rect, Size, PlaceholderAlignment;
 
 import 'package:flutter/foundation.dart';
 import 'package:flutter/gestures.dart';
 import 'package:flutter/services.dart';
 
 import 'basic_types.dart';
+import 'inline_span.dart';
 import 'strut_style.dart';
 import 'text_span.dart';
 
 export 'package:flutter/services.dart' show TextRange, TextSelection;
 
+/// Holds the [Size] and baseline required to represent the dimensions of
+/// a placeholder in text.
+///
+/// Placeholders specify an empty space in the text layout, which is used
+/// to later render arbitrary inline widgets into defined by a [WidgetSpan].
+///
+/// See also:
+///
+///  * [WidgetSpan] a subclass of [TextSpan] that represents an inline widget
+///    embedded within text. The space this widget takes is indicated by a
+///    placeholder.
+///  * [TextSpan] specifies the content. [WidgetSpan]s may be inserted as leaf
```

Comment:
"[TextSpan] specifies the content." is a weird sentence in this context. Not sure what it means.

Reply:
With all InlineSpans taking children, this is no longer relevant.